### PR TITLE
feat/tui-focus-mode — focus a single fleet, hide the rest

### LINKED ISSUE #169: Fleet TUI - focus mode.
# Overview
Add a new visual rendering mode to the fleet TUI main page. "Focus mode".  In focus mode only the specified fleet will render. Additionally the "fleet" ascii art logo will simplify to just a normal spelling of "Fleet".

## Motivation 

With many fleets configured the fleet TUI main page can consume a lot of vertical space. If a user is only working in a single fleet (which is very common) the rest of the space is being wasted. That space is better used for split panes which can show relivent information to the users task. 

## Workflow 

The user may select any fleet and hit the 'f' key to enter focus mode. When in focus mode all other fleets are hidden from the TUI. Only the "focused" fleet is shown. Additionally the "settings" option is replaced with "[ leave focus ]" option. Finally the "Fleet" ascii art is simplified to use the terminal quarter block characters, reducing vertical space while still making it pop out. The user can leave focus mode at any time by, 1. selecting the "[ leave focus ]" option and hitting enter, 2. hitting q when no such other element is present that would consume the q key (i.e dialogs). 3. hitting esc, basically the same as q. The focus state should basically be treated as a dialog when it comes to q / esc behavior.

### PR DESCRIPTION
**Pull request by BenjaminBenetti**
## Summary

Adds a **focus mode** to the fleet TUI main page (closes #169). With many fleets configured the list eats a lot of vertical space; when you're working in just one fleet, focus mode hides the rest so that space can go to split panes.

Press **`f`** on any fleet to focus it. While focused:

- **Only the focused fleet renders** — all other fleets are hidden.
- The **`settings`** list option is replaced with **`[ leave focus ]`**.
- The **help bar is hidden** (same effect as turning "show help text" off in settings) — focus mode is about reduced chrome, after all.
- **Leaving focus** works three ways, matching the ticket: select `[ leave focus ]` + enter, press `q`, or press `esc`. Focus behaves like a dialog for `q`/`esc` — they leave focus instead of quitting (`ctrl+c` still quits). Pressing `f` again also toggles it off.

> Note: the ticket originally asked for a compact quadrant-block "Fleet" banner in focus mode. That was implemented but reverted at the author's request (it didn't look good) — the standard banner is kept unchanged, and the freed intent (less chrome) is satisfied by hiding the help bar instead.

### Keybinding note

`f` was previously bound to **port-forward** on instance rows. To free `f` for focus (the ticket asks for `f`), **port-forward moved to `p`**. This also fixes a pre-existing inconsistency — `fleet port-forward` already told users to *"Use the TUI (press 'p')"* while the TUI actually used `f`. The contextual footer help and the Settings → Keybindings reference overlay are both updated.

## Changes

- `internal/tui/rows.go` — new `rowLeaveFocus` row kind.
- `internal/tui/page_fleet.go` — `focusedFleet` state; `buildRows` focus filtering + `[ leave focus ]` row; `enterFocus`/`leaveFocus`; `f`/`p`/`q`/`esc` handling (including the Armada-selector sub-state); help bar hidden in focus mode; `f: focus` / `p: port-forward` help hints.
- `internal/tui/keybindings.go` — reference overlay updated (`f: focus`, `p: port forward`).
- `internal/tui/focus_test.go` — unit tests for the state machine, key handling, render, help-hiding, and the Armada-selector edge cases.
- `integration/tests/340_tui_port_forward_dialog.sh` — drives the dialog with `p`.
- `integration/tests/345_tui_focus_mode.sh` *(new)* — end-to-end focus mode (enter, help hidden, leave via esc/q/leave-focus row).

### CODE REVIEW

File: internal/tui/page_fleet.go [hunk 1]
```diff
@@ -624,7 +715,7 @@ func (fleetPage *fleetPage) updateNormal(m *model, msg tea.Msg) tea.Cmd {
 						m.refreshSessionsFromRuntime(ref)
 						fleetPage.buildRows(m)
 					}
-				case rowSession, rowNewSession, rowSettings:
+				case rowSession, rowNewSession, rowSettings, rowLeaveFocus:
```
[the remark this answers is not in the document]
Comment (BenjaminBenetti) [reply internal/tui/page_fleet.go:718]: Thanks — I looked into this, but I don't think it's an issue: `A` is handled by its own top-level `case "A": return fleetPage.openArmadaSelect(m)` in the same `switch msg.String()`, which runs for every row (including `[ leave focus ]`). The `case " ", "tab":` block this comment is anchored to only matches Space/Tab, so `A` never reaches the row-kind sub-switch — pressing `A` opens the Armada selector as advertised. (Verified via the existing armada nav behavior; `A` isn't gated by row kind.)

**Comment by BenjaminBenetti** [comment]
/qa-manager Heads up on a behavior change since your last PASS (commit 83f733b) — please update the focus-mode plan accordingly:

The compact quadrant-block "Fleet" logo (your step #1) was **reverted at the author's request** — it looked bad. Focus mode now keeps the **standard banner unchanged**. To preserve the reduced-chrome intent, focus mode instead **hides the help bar** (the same effect as turning "show help text" off in settings).

So for the latest commit (47c672c), the expected focus-mode behavior is:
- Standard banner stays the same (no compact logo).
- \`settings\` → \`[ leave focus ]\`.
- **Help bar is hidden** while focused; it returns on leaving focus.
- Everything else (enter via f, leave via [ leave focus ]/q/esc, ctrl+c still quits, f toggles, p=port-forward) is unchanged.

Re-running QA via the ready-for-qa label now.
